src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectCreationExpression.cs

```diff
@@ -101,7 +101,31 @@ public override BoundNode VisitObjectCreationExpression(BoundObjectCreationExpre
                 return rewrittenObjectCreation;
             }
 
-            return MakeObjectCreationWithInitializer(node.Syntax, rewrittenObjectCreation, node.InitializerExpressionOpt, node.Type);
+            return MakeExpressionWithInitializer(node.Syntax, rewrittenObjectCreation, node.InitializerExpressionOpt, node.Type);
+        }
+
+        public override BoundNode VisitWithExpression(BoundWithExpression withExpr)
+        {
+            RoslynDebug.AssertNotNull(withExpr.CloneMethod);
+            Debug.Assert(withExpr.CloneMethod.ParameterCount == 0);
+
+            // for a with expression of the form
+            //
+            //      receiver with { P1 = e1, P2 = e2 }
+            //
+            // we want to lower it to a call to the receiver's `Clone` method, then
+            // set the given record properties. i.e.
+            //
+            //      var tmp = receiver.Clone();
+            //      tmp.P1 = e1;
+            //      tmp.P2 = e2;
+            //      tmp
+
+            return MakeExpressionWithInitializer(
+                withExpr.Syntax,
+                _factory.InstanceCall(VisitExpression(withExpr.Receiver), "Clone"),
```

Comment:
>"Clone" [](start = 74, length = 7)

moving earlier question here: I think we should use the information from binding (CloneMethod stored in bound node)